zipline/finance/trading.py

```diff
@@ -114,11 +124,11 @@ def __init__(
         if isinstance(asset_db_path, string_types):
             asset_db_path = 'sqlite:///%s' % asset_db_path
             self.engine = engine = create_engine(asset_db_path)
-            AssetDBWriterFromDictionary().init_db(engine)
         else:
             self.engine = engine = asset_db_path
 
         if engine is not None:
+            AssetDBWriter(engine).init_db()
```

Comment:
This was to make sure that we are always working with an initialized db. Now that I think about this, this should probably just be part of the constructor. How do you feel about that?